src/NLog/Internal/AssemblyHelpers.cs

```diff
@@ -102,9 +103,36 @@ public static Assembly LoadFromName(string assemblyName)
             Assembly assembly = assemblyPart.Load(stream.Stream);
             return assembly;
 #else
-            Assembly assembly = Assembly.Load(assemblyName);
-            return assembly;
+            try
+            {
+                Assembly assembly = Assembly.Load(AssemblyName.GetAssemblyName(assemblyName));
+                return assembly;
+            }
+            catch (FileNotFoundException)
+            {
+                var name = new AssemblyName(assemblyName);
```

Comment:
Maybe log a debug/Trace to the internal logger?

Reply:
Yep.